doc: clarify undocumented stream properties (#36715)
nodejs/node: doc/api/stream.md

faster or slower rate than data is received, each side should
operate (and buffer) independently of the other.

> The mechanics of the internal buffering are considered an internal

Reviewer: This and all subsequent suggestions are non-blocking/optional.

```suggestion
The mechanics of the internal buffering are an internal
```

> The mechanics of the internal buffering are an internal
> implementation detail and may be changed at any time. However, for certain
> advanced implementations, the internal buffers can be retrieved using
> `writable.writableBuffer` or `readable.readableBuffer`. Use of these undocumented
> properties is discouraged.

Author: ```suggestion
The mechanics of the internal buffering are an internal implementation detail
and may be changed at any time. However, for certain advanced implementations,
the internal buffers can be retrieved using `writable.writableBuffer` or
`readable.readableBuffer`. Use of these undocumented properties is discouraged.
```

> `writable.writableBuffer` or `readable.readableBuffer`, respectively. Use of
> these undocumented properties is not recommended.

Reviewer: There's no ordering that "respectively" is obviously referring to here, and the names make it clear what they are.

```suggestion
`writable.writableBuffer` or `readable.readableBuffer`. Use of
```

> these undocumented properties is not recommended.

Reviewer: ```suggestion
these properties is discouraged
```

Author: I prefer to keep the `undocumented` here because the specific properties themselves (and their values) are not documented.

> these undocumented properties is not recommended.

Author: ```suggestion
these undocumented properties is discouraged.
```

> `writable.writableBuffer` or `readable.readableBuffer`, respectively. Use of
> these undocumented properties is not recommended.

Author: ```suggestion
`writable.writableBuffer` or `readable.readableBuffer`. Use of these undocumented
properties is discouraged.
```
